bricks/strings/regex.h

```diff
@@ -0,0 +1,225 @@
+/*******************************************************************************
+The MIT License (MIT)
+
+Copyright (c) 2017 Dmitry "Dima" Korolev <dmitry.korolev@gmail.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+#ifndef BRICKS_STRINGS_REGEX_H
+#define BRICKS_STRINGS_REGEX_H
+
+#include <regex>
+#include <string>
+#include <unordered_map>
+
+namespace current {
+namespace strings {
+
+// A simple way to range-based-for-loop iterate over regex matches.
+struct IterateByRegexMatches {
```

Comment:
Nice.

Reply:
Thank you! :+1:

Reply:
Although in English it should be "range-based-for-loop-iterate", with the last `-` there as well, right?

Reply:
No. At least, I wouldn't include that last hyphen. The hyphen join of `range-base-for-loop` constructs a discreet modifier, which applies to the verb `iterate`

Reply:
And if "range based for loop iterate" is a verb, such as "to kickstart" or "to not-quite-agree"?

Reply:
Hard to say. Hyphen contraction is not a consistently standardized aspect of the English language. I think the relevant difference is: `range-based-for-loop iterate` is a noun-adjunct verb, like "eyeball-only estimate", rather than an adjective-adjunct verb, like "not-exactly-confirmed", but this is a language construct still in the midst of evolution.